editor.planx.uk/src/@planx/components/DrawBoundary/Public/index.tsx

```diff
@@ -134,6 +135,17 @@ export default function Component(props: Props) {
           newPassportData[PASSPORT_UPLOAD_KEY] = slots;
           newPassportData[PASSPORT_COMPONENT_ACTION_KEY] =
             DrawBoundaryUserAction.Upload;
+
+          // Track as requested file
+          const { required, recommended, optional } = passport.data?.[
+            PASSPORT_REQUESTED_FILES_KEY
+          ] || { required: [], recommended: [], optional: [] };
```

Comment:
I'm going to have another look at moving this repeated code fragment to the store / a shared location in another PR as my approach here was leading to failing tests https://github.com/theopensystemslab/planx-new/pull/2763/commits/f74f9da68edbce7406260051b8eb297fb21722ae

Reply:
PR here - #2770